### PR TITLE
[Add] get_orgs_for_permission

### PR DESCRIPTION
**Pull request by vgaviria**
## Summary: 
- adding `PermissionClient.get_orgs_for_permission` and
 `OsCoreServiceClient.fetch_orgs_for_permission`
## Test Plan:
 - [x] Modified Unit Tests

### CODE REVIEW

File: mbq/client/contrib/permissions.py [hunk 1]
```diff
@@ -64,6 +63,11 @@ def fetch_permissions_for_location(
     def fetch_all_permissions(self, person_id: str) -> FetchedPermissionsDoc:
         ...
 
+    def fetch_orgs_for_permission(
+        self, person_id: str, scope: str
+    ) -> Iterable[Union[UUIDType, str]]:
```
Comment (thieman) [inline mbq/client/contrib/permissions.py]: str is already in the type

Comment (thieman) [reply mbq/client/contrib/permissions.py]: also should we not make this always return str? we have full control over it so we can be more opinionated. the UUIDType is better for inputs

File: mbq/client/contrib/permissions.py [hunk 2]
```diff
@@ -427,3 +448,18 @@ def has_all_permissions(  # noqa: F811
             result=result,
         )
         return result
+
+    def get_orgs_for_permission(
```
Comment (thieman) [inline mbq/client/contrib/permissions.py]: Should the noun be `org_ref` instead? It's not always orgs

File: mbq/client/contrib/permissions.py [hunk 3]
```diff
@@ -17,7 +16,7 @@
 
 logger = logging.getLogger(__name__)
 
-UUIDType = Union[str, UUID]
+UUIDType = Union[str, uuid.UUID]
```
Comment (thieman) [inline mbq/client/contrib/permissions.py]: This is also the dumbest name I've ever come up with

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 *.py[co]
 .idea
 .DS_Store
+.mypy_cache/
```
Comment (johnheintschel1) [inline .gitignore]: wat dis is

File: mbq/client/contrib/permissions.py [hunk 5]
```diff
@@ -427,3 +448,18 @@ def has_all_permissions(  # noqa: F811
...
+        scope: str,
+    ) -> Iterable[str]:
+        """ Given a person and permission scope return all of the org or
+        location references where the person has that permission.
+        """
+        result = self.os_core_client.fetch_org_refs_for_permission(person_id, scope)
+        self.collector.increment(
+            "org_refs_for_permission",
+            tags={"call": "org_refs_for_permission", "result": str(result), "scope": scope},
+        )
```
Comment (johnheintschel1) [inline mbq/client/contrib/permissions.py]: can you please do a 

```self.registrar.emit("event_name_of_your_choice", person_id, scope, result=result) ```

:)

Comment (vgaviria) [reply mbq/client/contrib/permissions.py]: I did not see those before. Adding them now

**Pull request merged**